confluence-syntax-xhtml/src/main/java/org/xwiki/contrib/confluence/parser/xhtml/internal/wikimodel/URLTagHandler.java

```diff
@@ -53,6 +54,9 @@ protected void begin(TagContext context)
         WikiParameter urlParameter = context.getParams().getParameter("ri:value");
         if (urlParameter != null && container instanceof URLContainer) {
             ((URLContainer) container).setURL(urlParameter.getValue());
+        } else if (urlParameter != null && container instanceof ConfluenceMacro) {
+            ConfluenceMacro macro = (ConfluenceMacro) container;
+            macro.parameters = macro.parameters.setParameter("url", urlParameter.getValue());
```

Comment:
This arbitrary "url" parameter name does not make much sense. It's not the name of the parameter, it could collide with another one or simply imagine that you have several parameters of type URL.

Reply:
Ok, I get now that I probably misunderstood some things. Fixing it